packages/cli/src/cmds/assert/strategy/httpRequestStrategy.ts

```diff
@@ -0,0 +1,19 @@
+import Strategy from './strategy';
+import { AppMapData } from '../../../appland/types';
+import Assertion from '../assertion';
+
+export default class HttpRequestStrategy implements Strategy {
+  check(appMap: AppMapData, assertion: Assertion): Boolean {
+    for (let e of appMap.events) {
+      if (e.hasOwnProperty('http_server_response') && !eval(assertion.assert)) {
```

Comment:
Is it safe to `eval` like this? Are there better/safer ways?

Reply:
Generally I think we can consider this as trusted, peer reviewed code, but take a look at the [`vm` module](https://nodejs.org/api/vm.html#vm_vm_executing_javascript). `eval` has access to everything within the current scope and can introduce side effects as a result.

Reply:
> Generally I think we can consider this as trusted, peer reviewed code

That's what I had in mind too.

> `eval` has access to everything within the current scope and can introduce side effects as a result.

Right, we need to be careful about naming in our code.